packages/truffle-core/lib/commands/migrate.js

```diff
@@ -91,7 +91,7 @@ const command = {
     ]
   },
 
-  determineDryRunSettings: (config, options) => {
+  determineDryRunSettings: function(config, options) {
```

Comment:
I think this change is unnecessary but I suppose it doesn't hurt anything :)

Reply:
1. Keep the context syntax style consistent (same as +132)
2. In the following code( +184, +194), use `this.` call is not correct, you need to use `command.` to call, otherwise it will report the error and can't run.

Reply:
The solution I had was to pass the bound method to the deploy command. In that case `this` has the correct context.
This solution looks good to me. I have some basic test for these two commands at PR #2018